src/Util/MapUtil.js

```diff
@@ -0,0 +1,130 @@
+import OlMap from 'ol/map';
+import OlTileWMS from 'ol/source/tilewms';
+import OlTileLayer from 'ol/layer/tile';
+import OlTileGrid from 'ol/tilegrid/tilegrid';
+
+import { reverse, unionWith, isEqual, find, isEmpty } from 'lodash';
+
+import ObjectUtil from './ObjectUtil.js';
+
+import Logger from './Logger.js';
+
+/**
+ * Helper Class for the ol3 map.
+ *
+ * @class
+ */
+export class MapUtil {
+
+  /**
+   * Returns all interactions by the given name of a map.
+   *
+   * @param {ol.Map} map The map to use for lookup.
+   * @param {String} name The name of the interaction to look for.
+   * @return {ol.interaction[]} The list of result interactions.
+   */
+  static getInteractionsByName(map, name) {
+    let interactionCandidates = [];
+
+    if (!(map instanceof OlMap)) {
+      Logger.debug('Input parameter map must be from type `ol.Map`.');
+      return interactionCandidates;
+    }
+
+    let interactions = map.getInteractions();
+
+    interactions.forEach(function(interaction) {
+      if (interaction.get('name') === name) {
+        interactionCandidates.push(interaction);
+      }
+    });
+
+    return interactionCandidates;
+  }
+
+  /**
+   * Returns all interactions by the given name of a map.
+   *
+   * @param {ol.Map} map The map to use for lookup.
+   * @param {ol.interaction} clazz The class of the interaction to look for.
+   * @return {ol.interaction[]} The list of result interactions.
+   */
+  static getInteractionsByClass(map, clazz) {
+    let interactionCandidates = [];
+
+    if (!(map instanceof OlMap)) {
+      Logger.debug('Input parameter map must be from type `ol.Map`.');
+      return interactionCandidates;
+    }
+
+    let interactions = map.getInteractions();
+
+    interactions.forEach(function(interaction) {
+      if (interaction instanceof clazz) {
+        interactionCandidates.push(interaction);
+      }
+    });
+
+    return interactionCandidates;
+  }
+
+
+  /**
+   * @static buildLayersFromInitialState - Description
+   *
+   * @param {type} mapLayerObjArray Description
+   *
+   * @return {type} Description
+   */
+  static buildLayersFromInitialState(mapLayerObjArray) {
```

Comment:
Since this method parses the content of `mapLayers` delivered by SHOGun2 it can be discussed whether this method should be excluded here in favor of being a project specific helper method. If not, please add some description :wink: .